packages/hash/frontend/src/blocks/page/ComponentView.tsx

```diff
@@ -66,66 +76,64 @@ export const componentViewTargetSelector = "div[data-target=true]";
  */
 export class ComponentView implements NodeView<Schema> {
-  dom: HTMLDivElement = document.createElement("div");
-  contentDOM: HTMLElement | undefined = undefined;
-  editable: boolean;
+  public readonly dom = document.createElement("div");
+  public readonly contentDOM = document.createElement("div");
 
-  private target = createComponentViewTarget();
+  private readonly target = createComponentViewTarget();
 
-  private readonly componentId: string;
-  private readonly sourceName: string;
+  private readonly unsubscribe: Function;
 
   private store: EntityStore;
-  private unsubscribe: Function;
+
+  private wasSuggested = false;
 
   constructor(
     private node: ProsemirrorNode<Schema>,
-    public editorView: EditorView<Schema>,
-    public getPos: () => number,
-    private renderPortal: RenderPortal,
-    private meta: BlockMeta,
+    private readonly editorView: EditorView<Schema>,
+    private readonly getPos: () => number,
+    private readonly renderPortal: RenderPortal,
+    private readonly meta: HashBlockMeta,
   ) {
-    const { componentMetadata, componentSchema } = meta;
-    const { source } = componentMetadata;
-
-    if (!source) {
-      throw new Error("Cannot create new block for component missing a source");
-    }
-
-    this.sourceName = source;
-    this.componentId = componentMetadata.componentId;
-
     this.dom.setAttribute("data-dom", "true");
-
-    this.editable = blockComponentRequiresText(componentSchema);
-
-    if (this.editable) {
-      this.contentDOM = document.createElement("div");
-      this.contentDOM.setAttribute("data-contentDOM", "true");
-      this.contentDOM.style.display = "none";
-      this.dom.appendChild(this.contentDOM);
-    }
-
+    this.contentDOM.setAttribute("data-contentDOM", "true");
+    this.contentDOM.style.display = "none";
+    this.dom.appendChild(this.contentDOM);
     this.dom.appendChild(this.target);
 
+    this.dom.contentEditable = "false";
+
     this.store = entityStorePluginState(editorView.state).store;
     this.unsubscribe = subscribeToEntityStore(this.editorView, (store) => {
       this.store = store;
       this.update(this.node);
     });
 
+    this.wasSuggested =
+      suggesterPluginKey.getState(this.editorView.state)
+        ?.suggestedBlockPosition === this.getPos();
+
+    if (this.wasSuggested) {
+      this.editorView.dispatch(
+        this.editorView.state.tr.setMeta(suggesterPluginKey, {
+          type: "suggestedBlock",
+          payload: { position: null },
+        } as SuggesterAction),
+      );
+    }
+
     this.update(this.node);
   }
 
   update(node: ProsemirrorNode<Schema>) {
     this.node = node;
 
-    if (isComponentNode(node) && componentNodeToId(node) === this.componentId) {
-      const blockDraftId: string = this.editorView.state.doc
-        .resolve(this.getPos())
-        .node(2).attrs.draftId;
+    if (
+      isComponentNode(node) &&
+      componentNodeToId(node) === this.meta.componentId
+    ) {
```

Comment:
Prosemirror will sometimes call `update` on your `NodeView` with new nodes to see if they're compatible, so that they can be reused. If you return `false` from the `update` function, it will call `destroy` on your `NodeView` and create a new one instead. So this means in theory we could get `update` called with a component node representing a different component. This isn't supported, so we return false. See https://prosemirror.net/docs/ref/#view.NodeView.update

This is only something I know Prosemirror might do in theory, but haven't tested it. This logic is already there in main though - https://github.com/hashintel/hash/blob/main/packages/hash/frontend/src/blocks/page/ComponentView.tsx#L123

I'll add a comment explaining this.

Reply:
[`d28d807` (#872)](https://github.com/hashintel/hash/pull/872/commits/d28d80776defe42df585b607479ec9bb44d173b5)

Reply:
Understood, thank you